### ISSUE TITLE
Date format should use user language

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/deck#330]
We should use the users language to render the dates in the regional specific formats.

**Comment by juliushaertl** [comment]
Use the settings from https://github.com/nextcloud/server/pull/5623 once that is in.

**Comment by GoetheG** [comment]
Anything new on that topic here?

**Comment by willianveiga** [comment]
@juliushaertl, it's already using user's locale to display the dates.

The only piece that does not respect user's locale is the due date field value.

Am I missing something?

Thank you.

**Comment by juliushaertl** [comment]
@willianveiga Yes, that is exactly what this issue is about. The input field for due dates is now hardcoded to the YYYY-MM-DD but should use the users personal settings.

**Comment by juliushaertl** [comment]
@willianveiga I've invited you to the repo,  once you are a member, I can assign you. Great that you want to tackle that.

**Comment by willianveiga** [comment]
Great, @juliushaertl! Thank you very much! :smile:

**Comment by MathisP75** [comment]
It seems like this is broken as of 2026. It is now hardcoded to MM/DD/YYYY with 12H time, even when changing the locale.

<img width="359" height="55" alt="Image" src="https://github.com/user-attachments/assets/f70ff67a-49a5-44d1-8b47-01534cf5050b" />

**Comment by putt1ck** [comment]
@MathisP75 is that a view of the date shown on Deck card when selecting a due date? Looks fine on my installs

<img width="430" height="85" alt="Image" src="https://github.com/user-attachments/assets/02cb085b-84e1-43af-a2c6-f9db208c1601" />

Would be odd for a European built open source product to hardcode in the uniquely odd US date format.

**Comment by MathisP75** [comment]
Weirdly, it displays correctly on Chrome on my phone, but on Firefox on my desktop, it's unchanged. I tried on Chromium, and it's the same. Tried logging out, clearing cache, changing my browser's language, but it changes nothing. I have tried setting my locale in Nextcloud to UK English, Canadian French and others, but I see no changes on my desktop browser.

On Firefox, desktop:
<img width="1153" height="572" alt="Image" src="https://github.com/user-attachments/assets/a878c5a2-5fb3-4af8-af24-787b94c72e03" />

On Chrome, mobile:

<img width="1439" height="1794" alt="Image" src="https://github.com/user-attachments/assets/fdca015a-a4e8-4130-bafd-88e64266292e" />

**Comment by putt1ck** [comment]
I use FF so it seems something in your profile - you could  (non-destructively) test that by creating a new profile, make sure it's just got your preferred locale set as primary (I'd check in about:config and search for locale just to make it easy to see what's set and the ordering) and then visit your Deck install

**Comment by MathisP75** [comment]
It's not a Firefox issue. It does the same thing on Chromium. Also, it works fine on Firefox on my phone.

**Comment by putt1ck** [comment]
It's not an NC issue as such, because (a) it works elsewhere (I login to like 7 installations in various places, differing versions of NC/Deck, works in all those) and (b) it works fine on your phone. Do you have another PC you could try?

**Comment by MathisP75** [comment]
Ok, so it seems like Nextcloud will completely ignore your locale settings inside Nextcloud, and will instead rely on your browser's language. Mine was set to English US, so I set it to English CA and now it's fine.

**Comment by putt1ck** [comment]
Locale setting in NC controls the non-browser aspects e.g. email notifications
